src/main/java/org/apache/commons/collections4/list/AbstractLinkedList.java

```diff
@@ -387,7 +388,7 @@ public boolean equals(final Object obj) {
         while (it1.hasNext() && it2.hasNext()) {
             final Object o1 = it1.next();
             final Object o2 = it2.next();
-            if (!(o1 == null ? o2 == null : o1.equals(o2))) {
+            if (!(Objects.equals(o1, o2))) {
```

Comment:
How about in-lining `o1` and `o2` since they are not reused?

Reply:
Thanks for your suggestions, I will fix the commit right away.